src/card/interface/submitCardFields.js

```
@@ -33,6 +33,17 @@ type SubmitCardFieldsOptions = {|
   |}
 |};
 
+const convertMMYYToPaymentSourceDate = (dateString = "") => {
+  const dateArray = dateString.split('/');
+  const month = parseInt(dateArray[0] | "", 10);
```

Was this meant to be `dateArray[0] || ""`. However, `parseInt("", 10)` would return `NaN`, which would create an invalid date further down.

---

I'm going to write tests for this function and I can experiment with using strings vs numbers. I think the `Date` api expects `Date(number, number, number)` if you are going that route instead of passing a date string but i'm not sure

---

oh wait i see what you are saying now. we should just throw an error if we don't have a date string here

---

Yeah, if a thrown error would be handled correctly, then I think you could just do something like:
```js
  const month = parseInt(dateArray[0], 10);
  if (isNaN(month)) {
    // throw some error
  }
```

---

ended up getting rid of this completely